sklearn/utils/stats.py

```diff
@@ -1,18 +1,44 @@
 import numpy as np
 
 from .extmath import stable_cumsum
+from sklearn.utils.fixes import _take_along_axis
 
 
 def _weighted_percentile(array, sample_weight, percentile=50):
     """
     Compute the weighted ``percentile`` of ``array`` with ``sample_weight``.
+    If ``array`` is 2D, compute weighted ``percentile`` along axis=0.
     """
-    sorted_idx = np.argsort(array)
+    if sample_weight is None:
+        sample_weight = np.ones_like(array)
+    array = np.squeeze(array)
```

Comment:
Doing this seems inconsistent with the reference to axis=0 in the docstring, as any axis could be removed.

Reply:
Yes I understand your concern. The problem is that for single output the scorer function feeds `y_true` and `y_pred` as shape (n_samples, 1) `median_absolute_error` e.g., in `test_scorer_sample_weight`:
```
                weighted = scorer(estimator[name], X_test, target,
                                  sample_weight=sample_weight)
```
the shape of `y_pred` and `y_true` fed to `median_absolute_error` are (25, 1). This makes it difficult to differential between 1D array and multioutput 2D array, where you want the median value of each 'column'.

The only case I can see having trouble is where you have multioutput but only 1 same (1, n_labels). I could change to `np.squeeze(array, axis=-1)` ?

Reply:
That doesn't work. Did an if statement.